src/index/tests/index_tests.rs

```diff
@@ -1319,3 +1319,218 @@ fn sub_range_boundaries_are_registered_at_the_index() {
+    assert_eq!(data_type.initializer.as_ref(), initializer);
+}
+
+#[test]
+fn array_dimensions_are_stored_in_the_const_expression_arena() {
+    // GIVEN an array-datatype with constant expressions used in the dimensions
+    let src = "
+        TYPE MyInt : ARRAY[0 .. LEN-1, MIN .. MAX] OF INT;
+        ";
+    // WHEN the program is indexed
+    let (mut ast, ..) = crate::parser::parse(crate::lexer::lex(src));
+
+    crate::ast::pre_process(&mut ast);
+    let index = crate::index::visitor::visit(&ast);
+
+    // THEN I expect the index to contain constants expressions used in the array-dimensions
+
+    // check first dimensions 0 .. LEN-1
+    let (start_0, end_0) = index
+        .find_type_information("MyInt")
+        .map(|it| {
+            if let DataTypeInformation::Array { dimensions, .. } = it {
+                //return the pair (start, end)
+                (
+                    index
+                        .get_constant_expression(&dimensions[0].start_offset)
+                        .unwrap(),
+                    index
+                        .get_constant_expression(&dimensions[0].end_offset)
+                        .unwrap(),
+                )
+            } else {
+                unreachable!()
+            }
+        })
+        .unwrap();
+
+    assert_eq!(
+        format!("{:#?}", crate::parser::tests::literal_int(0)),
+        format!("{:#?}", start_0)
+    );
+
+    assert_eq!(
+        format!(
+            "{:#?}",
+            AstStatement::BinaryExpression {
+                id: 0,
+                operator: Operator::Minus,
+                left: Box::new(crate::parser::tests::ref_to("LEN")),
+                right: Box::new(crate::parser::tests::literal_int(1))
+            }
+        ),
+        format!("{:#?}", end_0)
+    );
+
+    //check 2nd dimension MIN .. MAX
+    let (start_1, end_1) = index
+        .find_type_information("MyInt")
+        .map(|it| {
+            if let DataTypeInformation::Array { dimensions, .. } = it {
+                //return the pair (start, end)
+                (
+                    index
+                        .get_constant_expression(&dimensions[1].start_offset)
+                        .unwrap(),
+                    index
+                        .get_constant_expression(&dimensions[1].end_offset)
+                        .unwrap(),
+                )
+            } else {
+                unreachable!()
+            }
+        })
+        .unwrap();
+
+    assert_eq!(
+        format!("{:#?}", crate::parser::tests::ref_to("MIN")),
+        format!("{:#?}", start_1)
+    );
+
+    assert_eq!(
+        format!("{:#?}", crate::parser::tests::ref_to("MAX")),
+        format!("{:#?}", end_1)
+    );
+}
+
+#[test]
+fn string_dimensions_are_stored_in_the_const_expression_arena() {
+    // GIVEN a string type with a const expression as length
+    let src = "
+        TYPE MyString : STRING(LEN-1);
```

Comment:
I think the we need STRING[] instead or you get a validation, not sure.
What happens to STRING[10]? Is it also added to the arena?

Reply:
ty done